src/rgw/rgw_lc.cc

```diff
@@ -0,0 +1,546 @@
+        if (prefix_match) {
+          days = match_days;
+        } else if (default_config) {
+          days = default_days;
+        } else {
+          continue;
+        }        
+        if (now - ceph::real_clock::to_time_t((*obj_iter).mtime) >= days*24*60*60) {
+          RGWObjectCtx rctx(store);
+          rgw_obj obj(bucket_info.bucket, (*obj_iter).key.name);
+          RGWObjState *state;
+          int ret = store->get_obj_state(&rctx, obj, &state, false);
+          if (ret < 0) {
+            return ret;
+          }
+          if (state->mtime != (*obj_iter).mtime) //Check mtime again to avoid delete a recently update object as much as possible
+            continue;
+          ret = rgw_remove_object(store, bucket_info, bucket_info.bucket, (*obj_iter).key);
+          if (ret < 0) {
+            ldout(cct, 0) << "ERROR: rgw_remove_object " << dendl;
+          } else {
+            ldout(cct, 10) << "DELETED:" << bucket_name << ":" << (*obj_iter).key.name <<dendl;
+          }
+        }
+      }    
+    } while (is_truncated);
+  } else {
+    for(map<string, int>::iterator prefix_iter = prefix_map.begin(); prefix_iter != prefix_map.end();  prefix_iter++) {
+      if (prefix_iter->first.empty()) {
+        continue;
+      }
+      list_op.params.prefix = prefix_iter->first;
+
+      do {
+
+        objs.clear();
+        list_op.params.marker = list_op.get_next_marker();          
+        ret = list_op.list_objects(1000, &objs, &common_prefixes, &is_truncated); 
+
+        if (ret < 0) {
+          if (ret == (-ENOENT))
+            return 0;
+          ldout(cct, 0) << "ERROR: store->list_objects():" <<dendl;
+          return ret;
+        }
+
+        vector<RGWObjEnt>::iterator obj_iter;
+        int days = prefix_iter->second;           
+        utime_t now = ceph_clock_now(cct);
+
+        for (obj_iter = objs.begin(); obj_iter != objs.end(); obj_iter++) {                  
+          if (now - ceph::real_clock::to_time_t((*obj_iter).mtime) >= days*24*60*60) {
+            RGWObjectCtx rctx(store);
+            rgw_obj obj(bucket_info.bucket, (*obj_iter).key.name);
+            RGWObjState *state;
+            int ret = store->get_obj_state(&rctx, obj, &state, false); 
+            if (ret < 0) {
+              return ret;
+            }
+            if (state->mtime != (*obj_iter).mtime)//Check mtime again to avoid delete a recently update object as much as possible
+              continue;
+            ret = rgw_remove_object(store, bucket_info, bucket_info.bucket, (*obj_iter).key);
+            if (ret < 0) {
+              ldout(cct, 0) << "ERROR: rgw_remove_object " << dendl;
+            } else {
+              ldout(cct, 10) << "DELETED:" << bucket_name << ":" << (*obj_iter).key.name << dendl;
+            }
+          }
+        }
+      } while (is_truncated);
+    }     
+  }
+
+  return ret;
+}
+
+int RGWLC::bucket_lc_post(int index, int max_lock_sec, cls_rgw_lc_obj_head& head, 
```

Comment:
This only returns 0, even on error.  If this is on purpose, why not a void return?